flink-contrib/docker-flink/Dockerfile

```diff
@@ -36,22 +31,24 @@ ENV PATH $PATH:$FLINK_HOME/bin
 EXPOSE 8081
 EXPOSE 6123
 
+# flink-dist can point to a directory or a tarball on the local system
+ARG flink_dist=NOT_SET
+
 # Install build dependencies and flink
+ADD $flink_dist $FLINK_INSTALL_PATH
 RUN set -x && \
-  mkdir -p $FLINK_INSTALL_PATH && \
-  apk --update add --virtual build-dependencies curl && \
-  curl -s $(curl -s https://www.apache.org/dyn/closer.cgi\?preferred\=true)flink/flink-${FLINK_VERSION}/flink-${FLINK_VERSION}-bin-hadoop${HADOOP_VERSION}-scala_${SCALA_VERSION}.tgz | \
```

Comment:
I see, just a terminology thing: image refers to the binary blob Docker Hub stores for download, in the Flink repo is a Dockerfile for generating that image. :)

This is mainly to follow suit with the other Apache projects I have found with official images; they all tend to host their Dockerfiles totally separately. Needless to say, the repo will be wholly licensed under the Apache license, just not maintained under the ASF.

And we're definitely committed to maintaining these official images going forward, updating them as a part of the Flink release process. If they need an individual maintainer point of contact, that can be me and I can hand it off later if I have to.